### ISSUE TITLE
2.8.0 on macos - error related to +UseCGroupMemoryLimitForHeap

### ISSUE THREAD

**Issue by pedro-w** [technomancy/leiningen#2329]
On macos High Sierra, Java 1.8.0_151-b12, Leiningen 2.8.0 I see warnings:
```
Java HotSpot(TM) 64-Bit Server VM warning: Unable to open cgroup memory 
limit file /sys/fs/cgroup/memory/memory.limit_in_bytes (No such file or directory)
```
I know about Issue #2321 that has been fixed but this is not quite the same; my JRE does have the `+UseCGroupMemoryLimitForHeap` option but macos doesn't have cgroups so the warning is printed. It is only a warning but still would be nice if it were fixed. I got my leiningen from Homebrew.

Steps to reproduce:
```
lein new app xyz
cd xyz
lein run
```

**Comment by winks** [comment]
@pedro-w Is that an Apple-supplied JVM or where did you get it?

**Comment by pedro-w** [comment]
@winks it's from the Oracle JDK; I downloaded and installed it yesterday to make sure I had the latest. (I am sticking with Java 8 for now because of a known incompatibility with something else on my system)

**Comment by pedro-w** [comment]
I checked out the repo referenced in #2330 and can confirm that the warning message no longer appears on macos. However, what would happen on other platforms that don't have cgroups  (e.g. Windows?)

**Comment by winks** [comment]
Thanks, I had found out about the downloading part meanwhile.

And yes, #2330 on Windows works, "os.name" is "Windows 7" for me, so the fn returns "true", as it does on Linux.

**Comment by winks** [comment]
OK, I effed up my test earlier.

I'll investigate later, but this is windows with some debug output:
```
λ c:\Apps\bin\leindev.bat version
Windows 7
true
Leiningen 2.8.1-SNAPSHOT on Java 1.8.0_144 Java HotSpot(TM) 64-Bit Server VM
```
=> no warning, but `use-cgroups-memory-limit-for-heap?` returns true

```
λ java -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -version
Java HotSpot(TM) 64-Bit Server VM warning: Unable to open cgroup memory limit file /sys/fs/cgroup/memory/memory.limit_in_bytes (No such file or directory)
java version "1.8.0_144"
Java(TM) SE Runtime Environment (build 1.8.0_144-b01)
Java HotSpot(TM) 64-Bit Server VM (build 25.144-b01, mixed mode)
```
echos the warning.

**Comment by pedro-w** [comment]
Not an expert on cgroups but as far as I can tell they are Linux only - so the test in `use-cgroups-memory-limit-for-heap?` should be a positive test for Linux, rather than a negative test for other platforms, shouldn't it? I suppose the BSDs are the only other family of OSs that are relevant.
Given that Oracle intend future versions of Java to transparently figure out if they are running in a container ([link]), in my humble opinion it would be better not to do this detection in Leiningen at all, but document where a user can put it in their `project.clj`/`profile.clj` if they need it.

[link]: https://blogs.oracle.com/java-platform-group/java-se-support-for-docker-cpu-and-memory-limits

**Comment by winks** [comment]
Didn't see your post earlier, redid the patch now to check for Linux.

I'm not happy with this solution either, but I think it *might* make sense to get a fixed 2.8.1 out of the door unless someone has an idea how to fix this properly.
